examples/timestamping/backend/tests/config.rs

```diff
@@ -73,24 +71,29 @@ fn init_testkit(second_time_service: bool) -> (TestKit, MockTimeProvider) {
     (testkit, mock_provider)
 }
 
-/// Creates block with `ConfigPropose` tx and returns `Result` with new configuration or corresponding `ExecutionError`.
-fn propose_configuration(testkit: &mut TestKit, config: Config) -> Result<Config, ExecutionError> {
-    let tx = ConfigPropose::immediate(0).service_config(SERVICE_ID, config);
+/// Creates block with `ConfigPropose` tx and returns `Result` with new
+/// configuration or corresponding `ExecutionError`.
+fn propose_configuration(testkit: &mut TestKit, config: Config) -> Result<(), ExecutionError> {
+    let tx = ConfigPropose::immediate(0).service_config(SERVICE_ID, config.clone());
 
     let (pub_key, sec_key) = testkit.network().us().service_keypair();
     let tx = tx.sign_for_supervisor(pub_key, &sec_key);
-    let block = testkit.create_block_with_transaction(tx.clone());
+    let block = testkit.create_block_with_transaction(tx);
 
-    if let Err(e) = block.index(tx.object_hash()).status() {
+    if let Err(e) = block[0].status() {
         return Err(e.clone());
     }
 
-    let new_config = testkit
+    let new_config: Config = testkit
         .api()
         .public(ApiKind::Service(SERVICE_NAME))
-        .get::<Config>("v1/timestamps/config")
+        .get("v1/timestamps/config")
         .expect("Failed to get service configuration");
-    Ok(new_config)
+    if config.time_service_name == new_config.time_service_name {
+        Ok(())
+    } else {
+        Err(Error::InvalidConfig.into())
```

Comment:
Is this return reachable? If the transaction has completed successfully, the configuration should be applied, shouldn't it? If so, I'd use

```rust
assert_eq!(config.time_service_name, new_config.time_service_name);
```